### ISSUE TITLE
Test-time Loss

### ISSUE THREAD

**Issue by cpauling** [open-mmlab/mmdetection#6749]
Hi,

I'm working on a project involving adversarial attacks on object detectors, and using MMDetection models. In order to implement the methods I am using, I need the loss of the predicted bboxes (and potentially other loss functions). Could you possibly explain how I could output a pre-trained model's loss, given an image at test-time?

Thank you.

**Comment by BIGWangYuDong** [comment]
https://github.com/open-mmlab/mmdetection/blob/f3817df55709e4b13098ebaa9cf72a77b68ce994/mmdet/models/detectors/two_stage.py#L129-L151
You can simple refer to `forward_train` to get losses

**Comment by BIGWangYuDong** [comment]
In mmdet, model(data) is different from PyTorch:
https://github.com/open-mmlab/mmdetection/blob/83eea4bec684708171f0b6433636df0873e50dac/mmdet/models/detectors/base.py#L157-L174

So, I don't think it can work

**Comment by cpauling** [comment]
It seems the main issue is that the input image tensor is not being stored in PyTorch's autograd graph. Given that the models use PyTorch, I am confused as to why this is happening. Could there be any way to manually ensure that the tensors are included (such as the `save_for_backward()` function) ?
